libgraphics/include/pivot/graphics/DescriptorAllocator/DescriptorLayoutCache.hxx

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <unordered_map>
+#include <vector>
+
+#include <vulkan/vulkan.hpp>
+
+namespace pivot::graphics
+{
+/// Cache the descriptor sets layout to avoid duplication
+class DescriptorLayoutCache
+{
+private:
+    struct DescriptorLayoutInfo {
+        std::vector<vk::DescriptorSetLayoutBinding> bindings;
+        bool operator==(const DescriptorLayoutInfo &other) const;
+        size_t hash() const;
+    };
+
+    struct DescriptorLayoutHash {
+        std::size_t operator()(const DescriptorLayoutInfo &k) const { return k.hash(); }
+    };
```

Comment:
What is the purpose of this struct ?

Reply:
It is used as the Hash operator for the map. I can't implement `std::hash`, because `DescriptorLayoutInfo` is private.